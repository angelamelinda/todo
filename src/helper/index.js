export function mockData() {
    return {
      // example data from redux store
      listTask:{
        
      },
  
      // actions
      RequestAddTask:jest.fn(),
      RequestEditTask:jest.fn(),
      RequestDeleteTask:jest.fn(),
      RequestChangeStatus:jest.fn(),
      RequestGetAllTask:jest.fn()
    }
  }
  