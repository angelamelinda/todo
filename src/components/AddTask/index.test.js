import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { AddTask } from './';
import * as helper from '../../helper';

Enzyme.configure({ adapter: new Adapter()});

describe('AddTask', () => {
    
    let props = helper.mockData();
    let mountedApp;

    const addTask = () => {
        if(!mountedApp) {
            mountedApp = mount (
                <AddTask {...props} />
            )
        }
        return mountedApp;
    }
    beforeEach(() => {
        mountedApp = null
    })

    it('handleSubmit when the button save is clicked', () => {
        let formAddTask = addTask().find('form');
        expect(formAddTask.exists()).toBe(true);
        formAddTask.find('input').instance().value = 'Senang yuk';
        formAddTask.simulate('submit');
        expect(addTask().prop("RequestAddTask")).toHaveBeenCalled();
    })
    
    it('matches with snapshot', () => {
        const wrapper = shallow(
            <AddTask {...props} />
        );
        expect(toJson(wrapper),{mode:'shallow'}).toMatchSnapshot();
    })
})

