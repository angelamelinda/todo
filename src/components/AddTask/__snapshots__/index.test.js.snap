// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask matches with snapshot 1`] = `
<div
  className="add-task"
>
  <form
    onSubmit={[Function]}
  >
    <input
      placeholder="What you want to do?"
      type="text"
    />
    <button
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;
